### ISSUE TITLE
Mask Branch

### ISSUE THREAD

**Issue by lcy0604** [hli2020/feature_intertwiner#2]
Hello, I have read your paper and feel really interested in your work. In your paper, you said that mask branch is not used. But when I run your code I found that mask_loss exists and mask branch seems to be still there in your code. Maybe I just get wrong but I am confused about this. 
I look forward to your reply. Thank you very much!

**Comment by hli2020** [comment]
Hi! Thanks for the interest! Yeah, in preliminary code there is a mask branch indeed. I will update this semi-pulic-ready codebase very soon.

**Comment by lcy0604** [comment]
> Hi! Thanks for the interest! Yeah, in preliminary code there is a mask branch indeed. I will update this semi-pulic-ready codebase very soon.

Thanks a lot! Sorry to bother you but I have another issue. I focus on scene text detection so that run your code with my own datasets. But I find that the loss of mrcnn_cls, mrcnn_bbox and mrcnn_mask are often 0, same as meta_loss. And the loss are always up and down. I just cannot figure it out. Do you encounter such a problem when on coco datasets?

**Comment by hli2020** [comment]
Hmm..that's weird. How do you split the reliable (large) /less reliable (small) sets? Similarly as we do for the anchors? The losses (cls, bbox, meta; ignore the mask loss) are definitely not zeros at first on COCO. They will decrease in a sharp way (eg, from 10.x --> 0.xxx in the first ~20k iters, if I remember correctly); then the loss would oscillate ("up and down" for the rest epoches), but AP keeps increasing as training goes. 

My guess would be credited to the dataset change. Maybe the text features for the big and small texts are similar. Not sure. How about change some `init` method for the model, `optim` method, `lr` etc? Or fix the input, the loss should for sure decrease right? Otherwise there must be some bug there.

**Comment by lcy0604** [comment]
> Hmm..that's weird. How do you split the reliable (large) /less reliable (small) sets? Similarly as we do for the anchors? The losses (cls, bbox, meta; ignore the mask loss) are definitely not zeros at first on COCO. They will decrease in a sharp way (eg, from 10.x --> 0.xxx in the first ~20k iters, if I remember correctly); then the loss would oscillate ("up and down" for the rest epoches), but AP keeps increasing as training goes.
> 
> My guess would be credited to the dataset change. Maybe the text features for the big and small texts are similar. Not sure. How about change some `init` method for the model, `optim` method, `lr` etc? Or fix the input, the loss should for sure decrease right? Otherwise there must be some bug there.

Emmm...I just change the data, anchor_ratios and the classes in your code. I check the format of the data and i think it is right. Maybe just as 

> Hmm..that's weird. How do you split the reliable (large) /less reliable (small) sets? Similarly as we do for the anchors? The losses (cls, bbox, meta; ignore the mask loss) are definitely not zeros at first on COCO. They will decrease in a sharp way (eg, from 10.x --> 0.xxx in the first ~20k iters, if I remember correctly); then the loss would oscillate ("up and down" for the rest epoches), but AP keeps increasing as training goes.
> 
> My guess would be credited to the dataset change. Maybe the text features for the big and small texts are similar. Not sure. How about change some `init` method for the model, `optim` method, `lr` etc? Or fix the input, the loss should for sure decrease right? Otherwise there must be some bug there.

**Comment by lcy0604** [comment]
> Hmm..that's weird. How do you split the reliable (large) /less reliable (small) sets? Similarly as we do for the anchors? The losses (cls, bbox, meta; ignore the mask loss) are definitely not zeros at first on COCO. They will decrease in a sharp way (eg, from 10.x --> 0.xxx in the first ~20k iters, if I remember correctly); then the loss would oscillate ("up and down" for the rest epoches), but AP keeps increasing as training goes.
> 
> My guess would be credited to the dataset change. Maybe the text features for the big and small texts are similar. Not sure. How about change some `init` method for the model, `optim` method, `lr` etc? Or fix the input, the loss should for sure decrease right? Otherwise there must be some bug there.

Emmm...I just change some default config such as classes and anchor_ratios. The data can be used on maskrcnn-benchmark in pytorch. I think the format of the datasets is right. Maybe just as you said, there are some details I don't know yet and I need to take a deeper insight on the work by myself. 
Thank you for your reply!
